kolibri/plugins/coach/class_summary_api.py

```diff
@@ -149,10 +152,27 @@ def get_num_correct(self, exam_log):
             .aggregate(Sum("correct"))
             .get("correct__sum")
         )
+    
+    def get_num_answered(self, exam_log):
+        #import pdb; pdb.set_trace()
+        return (
+            exam_log.attemptlogs.values_list("item")
+                .order_by("completion_timestamp")
+                .distinct()
+                .aggregate(
+                    complete__sum=Count(
+                        Case(
+                            When(complete=True, then=1),
+                            default=0
+                        )
+                    )
+                )
+                .get("complete__sum")
+        )
```

Comment:
note that this will get run for every quiz and every student in the class. Might want to do a quick performance check to make sure it's not too heavy

Reply:
Do you have any suggestions for how best to test the performance on this?

I reviewed the time it took for the `classSummary` api call to return in the browser - ~250ms with the code above in place.

When I change that function to return 0 - bypassing the DB work altogether - the time on that API call was +/- 20ms - not noticeably faster or slower.

I'm also working on a quiz with ~ 25 learners FWIW.

Reply:
okay - thanks for checking! Sounds like we're already fairly slow, but this doesn't change things much.

that process of testing sounds good to me, with the one caveat that the data might not be realistic. if you'd like to test with real data, aron or lingyi can give your our beta server DB.

Reply:
Thanks, @indirectlylit , I got the DB and tested some larger classrooms using the demo DB and didn't notice any significant change to the performance.

Reply:
As this function appears to be identical to the above, note that my suggestions there apply here too.

Another thing that occurs is that we should probably be enforcing this uniqueness at the DB level, rather than having to do the rather tenuous distinct workaround for SQLite that we need to do here (although that won't help with historical data, but would just be good to do for future data hygiene).

@indirectlylit the DRF list serializer issue (serializing every model in a queryset as an independently loaded entity) is also an issue here, and probably the biggest bottleneck to performance.

We could get a small speedup by doing a prefetch on the exam log query here:
https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/coach/class_summary_api.py#L283
```
        query_exam_logs = logger_models.ExamLog.objects.filter(
            exam__in=query_exams
        ).prefetch_related("attemptlogs").annotate(last_activity=Max("attemptlogs__end_timestamp"))
```